src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -330,73 +330,75 @@ def list_role_assignment_change_logs(cmd, start_time=None, end_time=None):  # py
     # Use the resource `name` of roleDefinitions as keys, instead of `id`, because `id` can be inherited.
     #   name: b24988ac-6180-42a0-ab88-20f7382dd24c
     #   id: /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c  # pylint: disable=line-too-long
-    role_defs = {d.name: worker.get_role_property(d, 'role_name') for d in list_role_definitions(cmd)}
-
-    for op_id in start_events:
-        e = end_events.get(op_id, None)
-        if not e:
-            continue
-
-        entry = {}
-        op = e.operation_name and e.operation_name.value
-        if (op.lower().startswith('microsoft.authorization/roleassignments') and e.status.value == 'Succeeded'):
-            s, payload = start_events[op_id], None
-            entry = dict.fromkeys(
-                ['principalId', 'principalName', 'scope', 'scopeName', 'scopeType', 'roleDefinitionId', 'roleName'],
-                None)
-            entry['timestamp'], entry['caller'] = e.event_timestamp, s.caller
-
-            if s.http_request:
-                if s.http_request.method == 'PUT':
-                    # 'requestbody' has a wrong camel-case. Should be 'requestBody'
-                    payload = s.properties and s.properties.get('requestbody')
-                    entry['action'] = 'Granted'
-                    entry['scope'] = e.authorization.scope
-                elif s.http_request.method == 'DELETE':
-                    payload = e.properties and e.properties.get('responseBody')
-                    entry['action'] = 'Revoked'
-            if payload:
-                try:
-                    payload = json.loads(payload)
-                except ValueError:
-                    pass
+    if start_events:
```

Comment:
Check if `start_events` has some value to reduce unnecessary calls to 
- Role Definitions API
- Graph Object API

to 
- Improve the performance of the command
- Reduce the size of recording YAML

Reply:
It seems that this change have nothing to do with the PR title, right ?

Reply:
It has no impact on the functionality either. If `start_events` is `[]`, the result of `list_role_definitions` will be discarded.